### DISCUSSION TITLE
Off-the-grid setup

### DISCUSSION THREAD

**Discussion by arne1921KF** [Nachtzuster/BirdNET-Pi#40] [Show and tell]
Hi there,

since [MueJosh](https://github.com/MueJosh) [asked over in the issue thread](https://github.com/Nachtzuster/BirdNET-Pi/issues/37#issuecomment-2059386347) alread, and @Nachtzuster is interested as well:

I'm currently trying to get my BirdNetPi off-grid. I have some hiccups, so nothing is finished, but I think I can start my thread here. This might even leverage my commitment to the project to a higher level.

So far, I have the following setup up & running:

- Raspberry Pi 4B 4 GB (with an additional HAT, see below) for about 40 €
- UGREEN USB-A exernal sound card (OEM SKU number 30724, chipset is a C-Media HS-100B) for about 6 € 
- SIGMA CM-20 microphone from the 1990s which was sitting in a drawer.

It's running bullseye. I switched to @Nachtzuster's branch based on a recommendation on the original repository. I am ever so thankful that Nachtzuster and y'all are engaging and friendly, and actively contributing to keep BirdNET-Pi alive. 

First thing to mention: at first, I had a terrible buzzing sound on the mic whenever using any plug-in power supply. With a power bank, this vanished. I was able to get rid of it by simply connecting GND to a radiator. (Caveat: This works in several European countries. Before you try, please  find out if your radiator has a potential equalisation connection.)

Second thing to mention:
I clicked my Pi on the secondary market, and it came with a HAT and no documentation whatsoever. All the seller contributed was that it had been used in a PV installation, and had thus an additional 6 to 24 V DC connection. I wondered if I could directly connect it to a solar panel, but I could not find out. The HAT is labeled as a Rocktech RPI_300_EXT_V1_0, and while I could find the company behind it, I could not find out *anything* about the board. I think it is a CAN bus I/O-board, and since I couldn't find out more, I am not going to try connecting the PV module directly at this moment.

But this got me thinking: well, let's get this thing to where the birds sing.

I started hawking for a couple of days, and got this:

- 20 W Solar Panel ("Topsolar", vendor E-CrossStu GmbH, 20Wp, 12V / 18Vmp, manufacturer Topunive Technology, Xiamen,)
- 10A 12V / 24V automatic PWM solar charge controller (with two USB-A ports) which came with the panel, both for under 30 € 
- 6 Ah LiFePO<sub>4</sub> battery (the brand is called Yazeky) with an unspecified battery management system for under 30 €

Right now, I can't test: the battery seems defective, my voltmeter shows 26 mV current. The case popped open by accident (it is neither welded nor glued shut), I got access to one of the four cells inside and did measure about 10 V current, so I guess the BMS is trying to protect the battery from getting deeply discharged. For me, it looks like it is already, so I will probably send it back.

To answer your question, @MueJosh: I took the next best bargain and clicked myself the panel including the PWM charging controller. After reading some of the advice on the web, I hope that that plus the BMS would be good enough for me. My reasoning is that, in my small setup with a 18V maxium, with (hopefully) trickle charging, MPPT will not be needed.  

Since the battery I got seems broken, I went slightly up-market and just ordered a battery from Eremit. They include a BMS which allows direct connection to a solar panel, without any additional charging controller. I got that as a tip from someone who has running systems of SBCs with  solar panels to record birds, bats, and insects. They also gave me the tip to go for the vendor Offgridtech for solar panels, and get a power management HAT to be able schedule recording times, to save battery. The latter I'm currently not willing to do, as it would be another 30 €. I'll have a look how far the battery will go, and come back to the idea of scheduling later.

As soon as by off-grid BirdNET-Pi is up and running, weather-restistant housing is going to be the next step.  

I'll try to keep you posted on my project. Feel free to ask if you want any details I forgot to mention.

**Comment by BAM-Team** [comment]
Hi @arne1921KF, @MueJosh, @Nachtzuster 
Great Project, thank you for sharing. We have a similar off Grid Station working for Biodiversity Monitoring. Check out Dirk´s Flickr Site: https://www.flickr.com/photos/143115576@N02/albums/72177720311660236/
Station has sometimes hickups -  RasPi hangups or Internet Problems. But with such cheap hardware, you can´t expect 24/7 
Block Diagram attached 
Kind Regards Carsten
P.S. Sim Card actually 27GB Data Rate
![Block Diagram](https://github.com/Nachtzuster/BirdNET-Pi/assets/164670893/ccdc0709-734f-45bb-b534-d070ec351065)

![IMG_8371](https://github.com/Nachtzuster/BirdNET-Pi/assets/164670893/0a561745-1d2e-47c1-a438-7c8614d4dc8f)

**Comment by krummrey** [comment]
Has anyone ever tried to go the simple route and build a just sound-logger?

No on site processing, just a simple device that will record sound for aprox. 24 hours. Detection then run on a desktop or cloud machine that just goes through the .WAVs collected by mentioned device the day before.

Maybe this ESP audio dev board + a battery would be sufficient? (currently around $15,- on aliexpress)
https://docs.ai-thinker.com/en/esp32-audio-kit

One could drop it off in the woods somewhere, retrieve it the next day...

**Reply by MueJosh** [reply]
> No on site processing, just a simple device that will record sound for aprox. 24 hours. Detection then run on a desktop or cloud machine that just goes through the .WAVs collected by mentioned device the day before.
> 
> 
> One could drop it off in the woods somewhere, retrieve it the next day...

You should definitely check this out! [microPAM](https://github.com/WMXZ-EU/microPAM)

An off the shelf alternative is the [audiomoth](https://www.openacousticdevices.info/audiomoth)

**Comment by arne1921KF** [comment]
So, after some hiccups with the battery, I am in the process to get my setup in a working condition.

I returned the first battery mentioned above and replaced it [with this one](https://www.eremit.de/p/12v-6ah-lifepo4-flach-mit-bms) - a 12 V LiFePO<sub>4</sub> with 6000 mAh. I guesstimated that it should bring the BirdNet-Pi through the night, but it did not.
Now, I'm trying to optimise my setup for maximum uptime and automatic restarting.

My current go at it: 
- trying to reduce power consumption of the Pi4B by running `sudo rpi-eeprom-config --edit` and setting `WAKE_ON_GPIO = 0` and `POWER_OFF_ON_HALT=1`
- creating a cronjob to shutdown at 22:00h at night ( i.e., running `sudo crontab -e` and editing in `00 20 * * * sudo poweroff`)
-  timing the PV panel's PWM charging controller for 18h of uptime (i.e., cutting current to the DC outlet after 18 hours of uptime)

My thinking is that the last part should, in theory, start the DC outlet again after six hours. I hope this starts the Pi, which then should run until ten pm. The battery, in theory, should last about 10h, and should get enough current during the day to stop the battery from draining, i.e. trickle charge throughout the day.

 Not to keep you waiting, here's my setup:
![birdnetbox-setup](https://github.com/Nachtzuster/BirdNET-Pi/assets/27221713/1d7ecf49-36b1-4b32-acc8-bf1c7c47187b)

As you can see, I dismantled the charging controller to save some space. The Pi itself is in a neat housing with a (sadly undocumented) HAT, which apparently provides a CAN bus interface. It also features a lithium battery, and I am rather sure there's a RTC on that board, but since I am new to hardware fiddling, I can't for the life of me figure out how to use it. However, I left it connected since it has a wide-range DC input on the back side of the case: I can run my Pi on 9 to 24 V. Which is exactly what I need - I could probably even connect the solar panel directly to it, since maximum output current is given as 18 V for this particular 20Wp panel, according to the specs on the label on it's back.

I set the charging controller to LiFePo<sub>4</sub>, and to use 14.4 V charging current, since the battery supports 14 to 16 V (it has it's own BMC, by the way, so in theory I could connect it directly to the panel). I also set the controller to 10 V output on the DC outlet, since I hope this a) saves some battery and b) reduces the overall amount of heat in the setup.  
![birdnetbox_interior](https://github.com/Nachtzuster/BirdNET-Pi/assets/27221713/e4a7ea48-9cfe-4f40-8183-6721ca6d3aea)
As you can see, my setup is a tight fit.  I had to force the lid shut.
![birdnetbox](https://github.com/Nachtzuster/BirdNET-Pi/assets/27221713/74f4137f-8279-4eb6-b6c4-9d09ae345db5)

As mentioned above, the 6000 mAh did not bring me through the night, so this is rather try and error. I think you need a bigger ~~boat~~ battery, if you want to build your own "offgrid" 24/7 BirdNET-Pi. If you do, please show and tell! 

Also, I'd be delighted if you'd share your expertise to get maximum uptime (i.e., how  you safe battery) and safe and proper shutting down(restarting of the BirdNET-Pi.

**Reply by arne1921KF** [reply]
Update: can't get it to get a reliable service. 

I set up a cron job to reboot in the morning and to halt in the evening, set the charging controller to 16h of service. Didn't work out as intended. 

Any suggestions?

**Reply by arne1921KF** [reply]
As indicated below, I am at a loss.

The Pi is possibly draining the 6 Ah LiFePo<sub>4</sub> faster than the 20Wp 12v PV panel can charge it. When the battery hits a threshold voltage, it shuts down (by internal BMS) or is shut down (by the charging controller). This means the charging controller _also_ shuts down. Which in turn means I have to disconnect the panel, the outlet and the battery (in that order) and reconnect them (inversely ordered) to get the bloody thing working again. 

If the battery isn't drained to much, that is: even during best weather conditions, I encountered glitches which brought the charging controller to something like a boot loop. It turned off and on continuously. When it turns on, the outlet is active, hence the Pi tries to start and the controller starts loading the battery. Then the controller shuts down, nearly immediately, disconnecting battery and Pi.

I have encountered this regardless of the value of the threshold set via the charging controller.

What @craigusus outlines below could possibly help: shutting the Pi down gracefully before the charging controller shuts down might just keep the system online to load the battery, and reboot the Pi once full capacity is reached. However, I think I would need a simple yet complete tutorial to be able to do that. 

I would wager at some point we'll have that. Until then, reader beware: don't make the same mistake as I did. Basically, choose a larger battery, and maybe a bigger PV panel.

¯\_(ツ)_/¯

**Reply by craigusus** [reply]
As you mentioned above, the Pi is draining your battery faster than the panel can charge it. Also, your charge controller will be using power from the battery to run, so that won't be helping the issue. This could be why it is going into a 'boot loop'.

- eg The panel charges the battery up enough to power on the controller > the controller (and Pi) turns on > the controller (and Pi) uses that little power and runs down the battery > the controller (and Pi) turns off > repeat.

Also, looking up the case you have (it took me a while), I have no idea what the board/HAT you have already, so that could also be consuming power. I would remove that. To be perfectly honest (I fully understand the cost implications) I would never buy a 2nd hand Pi as you never know what it was used for in its past. i.e. Bitcoin mining - running at 100% CPU, at high temperatures, 24/7/365 etc.

I think what you need for your setup is some sort of UPS HAT i.e. [waveshare ups hat](https://notenoughtech.com/raspberry-pi/waveshare-ups-hat) with the script adaption to tell the Pi to shut down (all UPS I have seen STILL do not have this feature!) when the UPS battery drops below a set voltage. Be wary of the 8.4V input to the UPS. You will also have to remove the Pi from your current case and HAT.

**Comment by craigusus** [comment]
Hi all,

I'm new to the gang and just wanted to show off my 'off-grid' setup.

My current hardware setup:
- Raspberry Pi 5 8GB
- [AGPTEK Lavalier Microphone](https://www.amazon.co.uk/dp/B07SHSHW6H)
- [KiwiBird USB to 3.5mm Jack Audio Adapter](https://www.amazon.co.uk/gp/product/B08JS9N98L) (ironically named)

After all the work-from-home malarky, stemming from COVID-19, I installed some DIY Solar Panels to power 2x 12V 100Ah batteries so that I could power my work laptop & screens etc on the days I work from home. I have 4x 100w panels in total.
I have fed (12V) power back from the charge controller to under my 2nd array of panels (that catch the noon-evening sun. 1st array catches the morning-afternoon sun) to the enclosure where I have my Pi 5. The 12V is connected to a buck converter stepping down to 5V, connected to the GPIO pins of the Pi.
![IMG20240527191416](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/e0a7b17e-e5a0-4c7c-b4c8-63aa6981d99d)
![IMG20240527191501](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/8a583693-bd21-4778-9efd-2e68cef3bea8)

I have mounted the enclosure underneath the panels, giving it a fair amount of weather protection (wind protection to be determined).

![IMG20240527191647](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/4487d34c-4e37-4264-978d-764ea4d1be25)
![IMG20240527191657_2](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/65e966a6-d017-44c3-924a-ba9da4adefa3)
![IMG20240527191704_2](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/d1cadb39-2550-495b-b395-6cc294d3c693)

From looking at my battery % over the past few nights, keeping the Pi running and charging my phone and watch overnight. It's only using about 2-3%. And even on an overcast day here in the UK (so 363 days of the year!) it will have topped that 2-3% up in a few hours.

ToDo/ThinkAbout:
- Add an automation to HomeAssistant to send a shutdown command to the Pi if the solar battery level drops below XX volts. I already have an ESP8266 running Tasmota, that monitors the battery voltage and sends that information via MQTT to HA.<img width="197" alt="image" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/c5ec03c6-60a1-4881-89b6-2d6b64d02373">
- Add an ESP8266 controlled relay to Pi's power supply so I can remotely toggle the power. Also, add control to HA.
- Add ventilation to the enclosure to reduce the Pi overheating in the summer
- Look at a small UPS style battery backup, similar to @arne1921KF's for if/when the solar battery is out of juice.

**Reply by arne1921KF** [reply]
Thank you for your post. 

Do you think it would be possible for you to determine the minimum PV panel wattage in combination with the minimum battery capacity to run your hardware continuously under real-world conditions?

I seem to have underestimated some factors in my setup, and I am heavily annoyed that my setup simply shuts down even during the day. I'll describe the behaviour above, so others can possibly learn from my trouble.

**Reply by craigusus** [reply]
No problem.

I can't with my current set-up as it is connected to my office, but I have been looking at getting a 50W panel to charge a spare 12V ~~100Ah~~ 20Ah to put into my shed. Once I get the panel I'll be able to do some tests.

I'll continue above...

**Reply by craigusus** [reply]
So I got paid, and the following day my 50W panel arrived 😉
I have just finished 'installing' it in its temporary location (on the floor, leaning up against my shed, facing 180° in the wrong direction!) with a cheapy controller (actually my first one when I started with solar panels a few years ago) and 1x 12V 20Ah battery. I'll test with 1 initially and possibly add the 2nd at a later date.

This has a Pi 4 1GB running off the USB port of the controller and a cheap USB webcam as the microphone.

According to the controller, it is receiving 0.2W and using 0W (somehow). 

I plan on putting the panel on the shed roof facing in the opposite direction (south) once the 2 trees you can see in the last picture are removed. Currently, they are housing a family of Common Wood-Pigeon's and it's illegal to tamper with the nests here, even though they are classed as vermin.

I'll see what the voltage is in a day or 2.

![IMG20240601165802](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/8ca1d86c-500c-45b3-8393-d59ba321abb2)
![IMG20240601165812](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/299f5635-33cc-4daf-a62b-c77532b128a9)
![IMG20240601165825](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/2b40454b-8ff9-4a4f-b474-e9ee2b271ca0)
![IMG20240601165840](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/6a4069b8-7a63-4b9b-8daf-aa5b1e25a1a1)
![IMG20240601170044](https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/db1e9f6b-e541-4a78-9a3b-3b983a76fe14)

**Reply by arne1921KF** [reply]
I've been busy for weeks, but coming back to off-grid planning I have to ask: how does your setup fare so far for true off-grid purposes?

**Reply by craigusus** [reply]
Hi, unfortunately not great. (I have been away this week, so everything was shut down).
I have been having power supply and solar controller issues. The Pi 5 was drawing more than the buck convertor could supply so the Pi kept crashing. And the controller in the shed kept failing and killing the power to the Pi 4.

I have now purchased a new [12V to 5V **15A**](https://www.amazon.co.uk/dp/B09PTPFYC7) and a new hinged [enclosure](https://www.amazon.co.uk/dp/B07RVN91WB) so I don't have to keep unscrewing the current one. But I have not installed them yet as Amazon delivered them late today. And I have a new controller for the shed. So I'm hoping to have better results this week!

**Reply by craigusus** [reply]
FINALLY! No more of these...
<img width="290" alt="image" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/755ce1a2-c72c-4c17-94d4-3f4f3818146f">
Still some more testing to do and all the cables are just shoved in the box for now (it should NOT have taken me 2 hours to do, yet it did!)
The Pi 5 is super nippy now too without the power drains.

I know it's only been a day but I thought I'd let you know the state of the shed batteries with the new charger.
I'm not too sure what went on, but my total voltage dropped to 1V between 1:30 am and 5 am. 2x 12V dropping to 1V, then ll the way back up to 12V within 3 hours?? Hmmm...
<img width="290" alt="image" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/a9117067-5e9b-4bcf-b38d-15bc01be56d8">

Current usage 0.4A is what the Pi 4 is pulling:
<img width="290" alt="image" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/a81c4428-462d-4970-a274-bfae8a99bd38">

Today's consumption 60Wh:
<img width="290" alt="image" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/6324863/61110886-a9a9-43d9-9516-16d8ecee6840">

I'll monitor for the next few days and keep you posted.

**Reply by arne1921KF** [reply]
@craigusus, any news?

Here: still experimenting with my 20Wp, my 6Ah LiFePO<sub>4</sub> plus a 20Ah standard Li-Ion power bank.

Using a wide-range (9-24V) DC input (with a HAT) supporting the Pi during the day. Charging the power bank via USB from the solar charging controller.  Using the 6Ah LiFePO<sub>4</sub> hooked up to the charging controller as buffer. Shutting down the Pi at sundown via a CRON job to reduce power draw. Shutting down the charging controller (via the internal timer, which sadly has no clock but just number of hours running) just before that to switch to USB-C power via the (now de-charging) power bank. Trying to reboot via CRON job at dawn. Experimenting with the timing of all of this does not get me anywhere. I suppose the 20Wp panel is just to weak to keep this running for longer than one night. 

How would you do troubleshooting with that setup? Any (software) suggestions for logging the state of the Pi easily, so I could try to optimise the setup? (Consider me a noob... the more verbose and on point your response, the better. ;-) )

**Comment by BAM-Team** [comment]
Great Setup, Craig 😃

I testet your Mic setup to get away a little bit cheaper than with my all -time favourite
Boya BY-LM40 USB-Lavalier-Mikrophone, but I found out Boya is stil better – so maybe you would give it a try…

Kind Regards, Carsten

**Comment by BAM-Team** [comment]
Addendum:

Agtek System: Today 384, Species detected today 15
Boya  System: Today 829, Species detected today 19

Identical Hardware orangepi02w, same local Mic Position etc.
Cheers, Carsten
